src/flow/coriolisterm_mod.F90

```diff
@@ -0,0 +1,197 @@
+
+    ! module functions
+    PUBLIC :: init_coriolisterm, finish_coriolisterm, coriolisterm
+
+CONTAINS
+
+    SUBROUTINE init_coriolisterm()
+
+        IF ( fort7%exists("/flow/coriolis") ) THEN
+
+            ! retrieving rotation rate vector from parameters.json
+            has_coriolis = .TRUE.
+            CALL fort7%get_array("/flow/coriolis/omega", omega)
+
+            ! display obtained parameters
+            IF (myid == 0) THEN
+                WRITE(*, '("CORIOLIS TERM:")')
+                WRITE(*, '(2X, "omega (rotation vector):   ", 3(G0, 1X))') omega
+                WRITE(*, '()')
+            END IF
+        END IF
+
+        RETURN
+
+    END SUBROUTINE init_coriolisterm
+
+
+    SUBROUTINE finish_coriolisterm
+
+        ! revoking activity
+        has_coriolis = .FALSE.
+
+        RETURN
+
+    END SUBROUTINE finish_coriolisterm
+
+
+    SUBROUTINE coriolisterm(uo_f, vo_f, wo_f)
+
+        ! Subroutine arguments (fields storing momentum balance)
+        TYPE(field_t), INTENT(inout) :: uo_f, vo_f, wo_f
+
+        ! Local variables
+        INTEGER(intk) :: i, igrid, kk, jj, ii
+        INTEGER(intk) :: nfro, nbac, nrgt, nlft, nbot, ntop
+        TYPE(field_t), POINTER :: u_f, v_f, w_f
+        REAL(realk), POINTER, CONTIGUOUS, DIMENSION(:, :, :) :: uo, vo, wo
+        REAL(realk), POINTER, CONTIGUOUS, DIMENSION(:, :, :) :: u, v, w
+
+        ! Checking activity
+        IF ( .NOT. has_coriolis ) RETURN
+
+        CALL start_timer(370)
+
+        ! getting the velocity fields
+        CALL get_field(u_f, "U")
+        CALL get_field(v_f, "V")
+        CALL get_field(w_f, "W")
+
+        ! iterating over all grids on processor (from grids_mod.F90)
+        DO i = 1, nmygrids
+            igrid = mygrids(i)
+
+            ! getting grid parameters (dimensions and boundary conditions)
+            CALL get_mgdims(kk, jj, ii, igrid)
+            CALL get_mgbasb(nfro, nbac, nrgt, nlft, nbot, ntop, igrid)
+
+            CALL uo_f%get_ptr(uo, igrid)
+            CALL vo_f%get_ptr(vo, igrid)
+            CALL wo_f%get_ptr(wo, igrid)
+
+            CALL u_f%get_ptr(u, igrid)
+            CALL v_f%get_ptr(v, igrid)
+            CALL w_f%get_ptr(w, igrid)
+
+            ! calling the function kernel for each grid
+            CALL coriolis_grid(kk, jj, ii, &
+                u, v, w, uo, vo, wo, &
+                nfro, nbac, nrgt, nlft, nbot, ntop)
+        END DO
+
+        CALL stop_timer(370)
+
+        RETURN
+
+    END SUBROUTINE coriolisterm
+
+
+    SUBROUTINE coriolis_grid(kk, jj, ii, &
+        u, v, w, uo, vo, wo, &
+        nfro, nbac, nrgt, nlft, nbot, ntop)
+
+        ! Subroutine arguments
+        INTEGER(intk), INTENT(in) :: kk, jj, ii
+        INTEGER(intk), INTENT(in) :: nfro, nbac, nrgt, nlft, nbot, ntop
+        REAL(realk), INTENT(in) :: u(kk, jj, ii), &
+            v(kk, jj, ii), w(kk, jj, ii)
+        REAL(realk), INTENT(inout) :: uo(kk, jj, ii), &
+            vo(kk, jj, ii), wo(kk, jj, ii)
```

Comment:
Following the style of mglet-base, the order of the arguments is to keep any argument you modify (i.e. intent OUT or INOUT) as early as possible in the argument list, and all pure inputs in the end.

The exception is the shape specification of any of the result/output parameters which goes first, even if the shape is intent(in) (always shape before array itself).

The always declare the arguments in the order of appearance.

```suggestion
    SUBROUTINE coriolis_grid(kk, jj, ii, &
        uo, vo, wo, u, v, w, &
        nfro, nbac, nrgt, nlft, nbot, ntop)

        ! Subroutine arguments
        INTEGER(intk), INTENT(in) :: kk, jj, ii
        REAL(realk), INTENT(inout) :: uo(kk, jj, ii), &
            vo(kk, jj, ii), wo(kk, jj, ii)
        REAL(realk), INTENT(in) :: u(kk, jj, ii), &
            v(kk, jj, ii), w(kk, jj, ii)
        INTEGER(intk), INTENT(in) :: nfro, nbac, nrgt, nlft, nbot, ntop
```